### ISSUE TITLE
`bundledDependencies` in yarn workspace

### ISSUE THREAD

**Issue by JakeLo123** [ng-packagr/ng-packagr#2853]
## `bundledDependencies` in yarn workspace

```
[x] Bug Report
[ ] Feature Request
```

## Description
I have an angular library that lives within a [yarn workspace](https://classic.yarnpkg.com/lang/en/docs/workspaces/). The library is dependent on a private package that lives in the same yarn workspace. I've included it in `bundledDependencies` in the library's package.json. My understanding is that the package code should be bundled into the build--however that's not the case. A module-not-found error occurs:

This comment mentions that the `embedded` api was replaced with `bundledDependencies`
https://github.com/ng-packagr/ng-packagr/issues/1106#issuecomment-425714633

#### How To Reproduce
I have created a repo that reproduces the issue: https://github.com/JakeLo123/test-monorepo2
- Clone the repo and run `yarn install`
- Build the library with `yarn build:lib` (this builds the core dependency, then the angular library)
- Inspect the outputted code in `packages/ng-ws/dist/user-lib/fesm2022/user-lib.mjs`

#### Expected Behaviour
The code from the `jakelo123-core` package should be bundled into the `mjs` file

#### Actual Behaviour
The code from `jakelo123-core` is imported as if it were a module dependency.

If you consume a local build of the library in an angular app, you will see the below error after starting the app. (I use [`yalc`](https://github.com/wclr/yalc) to consume the library locally)
```
Error: Module not found: Error: Can't resolve 'jakelo123-core' in '/Users/jakeloew/chalk/fusionauth/angular-quickstart/complete-application/node_modules/user-lib/fesm2022'
```

#### Version Information
```
ng-packagr: 17.3.0
@angular/compiler: 17.3.9
typescript: 5.3.3
node: 20.11.1
yarn: 1.22.19
```

**Comment by alan-agius4** [comment]
Since several major versions ago, ng-packagr no longer handles `bundledDependencies`. Instead, it relies on the package manager to include the bundled dependency as part of the distributable package.

I tried to replicate the resolution error using the provided reproduction, but I was unsuccessful.

**Comment by JakeLo123** [comment]
> I tried to replicate the resolution error using the provided reproduction, but I was unsuccessful.

Really? It bundled the `jakelo123` package into `packages/ng-ws/dist/user-lib/fesm2022/user-lib.mjs`?

**Comment by alan-agius4** [comment]
Bundling only happens when using `npm pack` or `npm publish`.

Can you please update the reproduction to show the `Module not found` error.

**Comment by JakeLo123** [comment]
> Bundling only happens when using `npm pack` or `npm publish`.

Ok, I see. So this is a no-go for my situation because the dependency--`jakeloew123-core`--is a private package, so installers of the library will get a 404 because it's dependency, and not listed on the NPM registry.

> Can you please update the reproduction to show the `Module not found` error.

The reproduction contains the `Module not found` error under "Actual Behaviour", no? Please let me know if you'd like me to edit that in a specific way, I'm happy to.

...so it sounds like I have incorrectly filed this issue as a bug. `ng-packagr` does not bundle dependencies. If that's the case, I can close this ticket 👍

**Comment by alan-agius4** [comment]
> Ok, I see. So this is a no-go for my situation because the dependency--jakeloew123-core--is a private package, so installers of the library will get a 404 because it's dependency, and not listed on the NPM registry.

They will not as the when using `bundledDependencies` it will be included as part of the NPM package. NB: this dependency should not be listed in the dependencies section.

I’ll check the reproduction once again in the coming days.

**Comment by JakeLo123** [comment]
Yes you're right, it should be in `devDependencies` not `dependencies`--nonetheless, it looks like I'll have to either publish the core package and consume it like a regular dependency or consider a different approach to building this. Thanks for your quick replies.
